### DISCUSSION TITLE
Errors pending resolution for months and question about the future of the project

### DISCUSSION THREAD

**Discussion by manelio** [TanStack/router#617] [General]
With all due respect and admiration for the great work with the Tanstack libraries, the case with Router concerns me. I started a significant project with Location, which had some major errors, though the development is running in production without issues. For the next version of the project, I waited over a year for Router to be minimally stable. I can deal with some problems and errors, but I expected a basic stable functionality.

I have created a very small codesandbox with the latest version 0.0.1-beta.89:

https://codesandbox.io/s/modest-wood-h6g7gg

- The library is totally unusable due to the issue reported at https://github.com/TanStack/router/issues/597 50 days ago. Clicking on a link is always one state behind.

- Route matching does not work correctly with routes that share a prefix as in:

```
/posts/$postId
/posts
```

and to make it work it is necessary to wrap the routes in a common route.

Reported three months ago, on March 20: https://github.com/TanStack/router/issues/562

- The same happens with wildcard routes. It's impossible to detect a route change, which breaks a very common pattern such as loading generic pages with information received from the server, for example.

This problem is reported for Location over 8 months ago (https://github.com/TanStack/router/issues/184), closed with the comment *This is both stale and fixed in the latest Router version*. (https://github.com/TanStack/router/issues/184#issuecomment-1310392913)

But as can be seen in the codesandbox example, the loader for the route

```
const catchAllRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "*",
  loader: async ({ params }) => {
    console.log(`Load ${params["*"]}`);
    return params;
  },
  component: () => {
    return <h1>Some page</h1>;
  }
});
```

only runs once, the first time you click on a link that matches the wildcard.

I understand that developing an open-source project is hard and costly. This is a project with a very good look, but I would like a sincere response about the intentions to evolve and stabilize the project. So that I can make a decision about whether to switch to another library, wait, or actively collaborate by trying to resolve some of the errors.

I think that by solving the errors that affect basic functionality, it would allow this library to be used in a large percentage of cases where it is now unusable.

For now, I am going to try to help by creating a fork and trying to solve some of the errors. If I succeed, I will make a pull request.

**Comment by tannerlinsley** [comment]
Hey @manelio, thanks for your concern. To justify the slowdown, I've been in 2 parallel product sprints at Nozzle for the last few months and haven't had time for OSS much. I was clear about this hiatus on our discord, so you must have missed it. That's fine. I'm currently swinging back into things (our product launches went wonderfully). Despite getting my tonsils out and being loopy for the last week, I managed to fix a few of the critical issues you were experiencing and am working on more.

I don't feel you'll need to resort to a fork, but I'm flattered you would :) Chat with me more on discord if you have more questions/concerns.
